draft-ietf-oauth-v2-1.md

```diff
@@ -1062,31 +1042,200 @@ request using a pre-defined default value or fail the request
+           client authentication included, or unsupported
+           authentication method).  The authorization server MAY
+           return an HTTP 401 (Unauthorized) status code to indicate
+           which HTTP authentication schemes are supported.  If the
+           client attempted to authenticate via the `Authorization`
+           request header field, the authorization server MUST
+           respond with an HTTP 401 (Unauthorized) status code and
+           include the `WWW-Authenticate` response header field
+           matching the authentication scheme used by the client.
+
+     "invalid_grant":
+     :     The provided authorization grant (e.g., authorization
+           code, resource owner credentials) or refresh token is
+           invalid, expired, revoked, does not match the redirect
+           URI used in the authorization request, or was issued to
+           another client.
+
+     "unauthorized_client":
+     :     The authenticated client is not authorized to use this
+           authorization grant type.
+
+     "unsupported_grant_type":
+     :     The authorization grant type is not supported by the
+           authorization server.
+
+     "invalid_scope":
+     :     The requested scope is invalid, unknown, malformed, or
+           exceeds the scope granted by the resource owner.
+
+     Values for the `error` parameter MUST NOT include characters
+     outside the set %x20-21 / %x23-5B / %x5D-7E.
+
+"error_description":
+:    OPTIONAL.  Human-readable ASCII [USASCII] text providing
+     additional information, used to assist the client developer in
+     understanding the error that occurred.
+     Values for the `error_description` parameter MUST NOT include
+     characters outside the set %x20-21 / %x23-5B / %x5D-7E.
+
+"error_uri":
+:    OPTIONAL.  A URI identifying a human-readable web page with
+     information about the error, used to provide the client
+     developer with additional information about the error.
+     Values for the `error_uri` parameter MUST conform to the
+     URI-reference syntax and thus MUST NOT include characters
+     outside the set %x21 / %x23-5B / %x5D-7E.
+
+The parameters are included in the payload of the HTTP response
+using the `application/json` media type as defined by [RFC7159].  The
+parameters are serialized into a JSON structure by adding each
+parameter at the highest structure level.  Parameter names and string
+values are included as JSON strings.  Numerical values are included
+as JSON numbers.  The order of parameters does not matter and can
+vary.
+
+For example:
+
+    HTTP/1.1 400 Bad Request
+    Content-Type: application/json
+    Cache-Control: no-store
+
+    {
+     "error":"invalid_request"
+    }
+
+# Grant Types {#obtaining-authorization}
+
+To request an access token, the client obtains authorization from the
+resource owner. OAuth defines two authorization grant types: authorization code
```

Comment:
This needs to be updated to include the refresh token grant type

Reply:
fixed